models/weight_recorder.h

```diff
@@ -131,7 +131,19 @@ class weight_recorder : public Node
   bool user_set_precise_times_;
   bool has_proxies_;
   bool local_receiver_;
-  omp_lock_t writelock;
+
+  struct Parameters_
+  {
+    std::set< long > pre_;
+    std::set< long > post_;
```

Comment:
I think `set` is a good first choice, we can later consider if we need something even more efficient. Note that we usually use the alias `index` for GIDs, I'd suggest using that here as well.